code/obj/flock/structure/gnesisturret.dm

```diff
@@ -188,3 +189,18 @@
 		. = ..()
 		parentTurret.reagents.trans_to(O,src.cost)
 		parentTurret.info_tag.set_info_tag("Gnesis: [parentTurret.reagents.total_volume]/[parentTurret.fluid_level_max]")
+
+/obj/item/implant/projectile/body_visible/syringe/syringe_barbed/gnesis_nanites
+	name = "barbed crystaline spike"
+	desc = "A hollow teal crystal, like some sort of weird alien syringe. It has a barbed tip. Nasty!"
+	New()
+		. = ..()
+		src.material = getMaterial("gnesis")
+
+	on_life(mult)
+		. = ..()
+		if(src.reagents?.total_volume == 0)
+			if(!ON_COOLDOWN(src.owner, "gnesis_barb_spam", 5 SECONDS))
```

Comment:
Is cooldown needed if its checking for volume of 0?

Reply:
the cooldown is on the mob, to prevent multiple darts decaying at once spamming the chat.